src/visualization/components/internal/LegendOptions.tsx

```diff
@@ -130,34 +134,37 @@ export const OrientationToggle: FC<OrientationToggleProps> = ({
   )
 }
 
-export const OpacitySlider: FC<OpacitySliderProps> = ({
-  legendOpacity,
-  handleSetOpacity,
-  testID = 'opacity-slider',
-}) => {
+export const OpacitySlider: FC<OpacitySliderProps> = props => {
   let validOpacity = LEGEND_OPACITY_DEFAULT
   if (
-    typeof legendOpacity === 'number' &&
-    legendOpacity === legendOpacity &&
-    legendOpacity >= LEGEND_OPACITY_MINIMUM &&
-    legendOpacity <= LEGEND_OPACITY_MAXIMUM
+    typeof props.legendOpacity === 'number' &&
+    props.legendOpacity === props.legendOpacity &&
+    props.legendOpacity >= LEGEND_OPACITY_MINIMUM &&
+    props.legendOpacity <= LEGEND_OPACITY_MAXIMUM
   ) {
-    validOpacity = legendOpacity
+    validOpacity = props.legendOpacity
   }
-  const percentLegendOpacity = (validOpacity * 100).toFixed(0)
+  const [opacity, setOpacity] = useState(validOpacity)
+  const setOpacityDebounced = useCallback(debounce(props.setOpacity, 350), [])
```

Comment:
since we have a `setOpacity` functin from the `useState` hook and a `props.setOpacity`, rather than destructuring `props` and dealing with a naming collision, just use the `props` scope to refer to the version of `setOpacity` that gets passed in